keras_cv/models/resnet.py

```diff
@@ -0,0 +1,547 @@
+            specified.
+        weights: one of `None` (random initialization), or a pretrained weight file
+            path.
+        input_shape: optional shape tuple, defaults to (None, None, 3).
+        pooling: optional pooling mode for feature extraction
+            when `include_top` is `False`.
+            - `None` means that the output of the model will be the 4D tensor output
+                of the last convolutional block.
+            - `avg` means that global average pooling will be applied to the output
+                of the last convolutional block, and thus the output of the model will
+                be a 2D tensor.
+            - `max` means that global max pooling will be applied.
+        name: (Optional) name to pass to the model.  Defaults to "{name}".
+    Returns:
+      A `keras.Model` instance.
+"""
+
+
+def ResNet(
+    stack_fn,
+    use_preactivation,
+    include_rescaling,
+    name="ResNet",
+    include_top=True,
+    weights="None",
+    input_shape=(None, None, 3),
+    pooling=None,
+    num_classes=1000,
+    classifier_activation="softmax",
+    **kwargs,
+):
+    """Instantiates the ResNet and ResNetV2 architecture.
+
+    Args:
+      stack_fn: a function that returns output tensor for the
+        stacked residual blocks.
+      use_preactivation: whether to use pre-activation or not
+        (True for ResNetV2, False for ResNetV1).
+      include_rescaling: whether or not to Rescale the inputs. If set to True,
+        inputs will be passed through a `Rescaling(1/255.0)` layer.
+      name: string, model name.
+      include_top: whether to include the fully-connected
+        layer at the top of the network.
+      weights: one of `None` (random initialization),
+        or the path to the weights file to be loaded.
+      input_shape: optional shape tuple, defaults to (None, None, 3).
+      pooling: optional pooling mode for feature extraction
+        when `include_top` is `False`.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional layer.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional layer, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+      num_classes: optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+      **kwargs: For backwards compatibility only.
```

Comment:
In the VGG and DenseNet impls in keras_cv, we use kwargs as a pass-through for params to Keras.model(). I think we should remain consistent with that here.

Either way, I don't think we need to include them for backwards-compatability as this is being moved to a new namespace.